gateway/src/main/scala/filodb/gateway/GatewayServer.scala

```diff
@@ -129,14 +131,22 @@ object GatewayServer extends StrictLogging {
     val genHist = userOpts.genHistData.getOrElse(false)
     val genGaugeData = userOpts.genGaugeData.getOrElse(false)
     val genDeltaHist = userOpts.genDeltaHistData.getOrElse(false)
-    if (genHist || genGaugeData || genDeltaHist) {
+    val genCounterData = userOpts.genCounterData.getOrElse(false)
+    val genDeltaCounterData = userOpts.genDeltaCounterData.getOrElse(false)
+    if (genHist || genGaugeData || genDeltaHist
+          || genCounterData || genDeltaCounterData) {
       val startTime = System.currentTimeMillis
       logger.info(s"Generating $numSamples samples starting at $startTime....")
 
       val stream = if (genHist) TestTimeseriesProducer.genHistogramData(startTime, numSeries, promHistogram)
                    else if (genDeltaHist) TestTimeseriesProducer.genHistogramData(startTime, numSeries, deltaHistogram)
-                   else TestTimeseriesProducer.timeSeriesData(startTime, numSeries,
-                                         userOpts.numMetrics(), userOpts.publishIntervalSecs())
+                   else if (genGaugeData) TestTimeseriesProducer.timeSeriesData(startTime, numSeries,
```

Comment:
This is internally using PrometheusInputRecord, which is writing data in `untyped` schema, instead of `gauge` schema. Maybe we would need another InputRecord type for gauge as well ?

Reply:
I will fix this in another PR